llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.h

```diff
@@ -240,6 +240,11 @@ class GCNTTIImpl final : public BasicTTIImplBase<GCNTTIImpl> {
   bool areInlineCompatible(const Function *Caller,
                            const Function *Callee) const;
 
+  int getInliningLastCallToStaticBonus() const {
+    // InlineConstants::LastCallToStaticBonus * 11 where 11 is AMDGPU's inlining
+    // threshold multiplier.
+    return 165000;
```

Comment:
Can you just take the existing multiplier and apply it to the existing threshold without the new layer? I would have guessed the multiplier applying to bonuses and costs would cancel out

Reply:
> Can you just take the existing multiplier and apply it to the existing threshold without the new layer?

That is what currently is doing but apparently the threshold is not large enough for a large function even after the constant bonus applied.

Reply:
The multiply is not applied to the LastCallToStaticBonus below

Reply:
That is gonna change all the targets, which I'm not sure if that's a good idea. That is also the motivation to have a target dependent API instead of assuming all targets want to inline the last call.

Reply:
It makes no sense to have this multiplier applied to some places and not the others. Do not do things out of fear of touching other targets, that's how we end up with a lot of the messes we already have

Reply:
I applied the multiplier just by choice. It is not like the general model should apply it. I can just set a random large number. There is no general rule (or studied heuristic) here, which exactly echoes the need for a target dependent bonus.

Reply:
Just like the threshold multiplier hook we already have. It's basically the do GPU inlining flag already

Reply:
So you were suggesting to add a bonus multiplier?

Reply:
I mean it's preferable to just implement this in terms of what we already have. If the multiplier is already applied to other bonuses, and not this bonus, it's easier and more consistent to just apply it there too.

Reply:
I don't think the multiplier was already applied to other bonuses. For example, `VectorBonus` has a TTI function `getInlinerVectorBonusPercent` to determine the "multiplier".